### PR TITLE
update saver to take args

### PR DESCRIPTION
**Pull request by Ssukriti**
**What this PR does / why we need it**:
Both watson_nlp and caikit_nlp pass args to save in certain modules. 

**Special notes for your reviewer**:
Blocking other efforts, kindly prioritize
 
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
Thanks Sukriti! Can you throw a unit test in somewhere to ensure that this works end-to-end so that we don't accidentally break this in the future?

File: tests/core/modules/test_module.py [hunk 1]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
     dummy_model.run_batch(sample_input=[SampleInputType(name="Gabe")])
 
 
+## ModuleSaver ################################################################
+
+
+def test_save_module(model_path):
```
Comment (Ssukriti) [inline tests/core/modules/test_module.py:140]: just FYI, these are taken from watson_core

File: tests/core/modules/test_module.py [hunk 2]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+    dummy_block = caikit.core.load(model_path)
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:141]:
```suggestion
    dummy_model = caikit.core.load(model_path)
```

File: tests/core/modules/test_module.py [hunk 3]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with ModuleSaver(
+            dummy_block,
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:145]:
```suggestion
            dummy_model,
```

File: tests/core/modules/test_module.py [hunk 4]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+            model_path=tempdir,
+        ) as module_saver:
+            module_saver.save_module(dummy_block, "dummy")
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:148]:
```suggestion
            module_saver.save_module(dummy_model, "dummy")
```

File: tests/core/modules/test_module.py [hunk 5]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+            assert os.path.exists(os.path.join(tempdir, "dummy"))
+            assert module_saver.config.get("module_paths") is not None
+            assert module_saver.config.get("module_paths") == {"dummy": "./dummy"}
+
+
+def test_save_module_kwargs_get_piped_through(model_path):
+    """Testing additional keywords passed to save_module are processed by the module"""
+    dummy_block = caikit.core.load(model_path)
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:157]:
```suggestion
    dummy_model = caikit.core.load(model_path)
```

File: tests/core/modules/test_module.py [hunk 6]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with ModuleSaver(
+            dummy_block,
+            model_path=tempdir,
+        ) as module_saver:
+            module_saver.save_module(dummy_block, "dummy", keyword=True)
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:164]:
```suggestion
            module_saver.save_module(dummy_model, "dummy", keyword=True)
```

File: tests/core/modules/test_module.py [hunk 7]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+            saved_block = caikit.core.load(os.path.join(tempdir, "dummy"))
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:165]:
```suggestion
            saved_model = caikit.core.load(os.path.join(tempdir, "dummy"))
```

File: tests/core/modules/test_module.py [hunk 8]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+            assert saved_block.metadata["test_keyword"]
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:166]:
```suggestion
            assert saved_model.metadata["test_keyword"]
```

File: tests/core/modules/test_module.py [hunk 9]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+
+def test_save_module_list(model_path):
+    """Test valid module list save with a list of syntax blocks."""
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:170]:
```suggestion
    """Test valid module list save with a list of models."""
```

File: tests/core/modules/test_module.py [hunk 10]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+    dummy_block = caikit.core.load(model_path)
+    dummy_models_with_rel_path = {"dummy_path": dummy_block}
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:172]:
```suggestion
    dummy_models_with_rel_path = {"dummy_path": dummy_model}
```

File: tests/core/modules/test_module.py [hunk 11]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with ModuleSaver(
+            dummy_block,
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:176]:
```suggestion
            dummy_model,
```

File: tests/core/modules/test_module.py [hunk 12]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+            saved_subpath = os.path.split(list_of_rel_path[0])
+            assert len(saved_subpath) == 2
+            # ...Where the first resolves to the relative path [no hierarchy expected here]
+            assert not saved_subpath[0].strip(".")
+            # ...And the second is the dummy_path we specified
+            assert saved_subpath[1] == "dummy_path"
+
+
+def test_save_module_list_kwargs_get_piped_through(model_path):
+    """Check that additional keywords passed to save_module_list are processed by the block"""
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:194]:
```suggestion
    """Check that additional keywords passed to save_module_list are processed by the model"""
```

File: tests/core/modules/test_module.py [hunk 13]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+    dummy_block = caikit.core.load(model_path)
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:195]:
```suggestion
    dummy_model = caikit.core.load(model_path)
```

File: tests/core/modules/test_module.py [hunk 14]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+    dummy_models_with_rel_path = {"dummy_path": dummy_block}
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:196]:
```suggestion
    dummy_models_with_rel_path = {"dummy_path": dummy_model}
```

File: tests/core/modules/test_module.py [hunk 15]
```diff
@@ -134,6 +134,79 @@ def test_run_batch_keyword_only(model_path):
...
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with ModuleSaver(
+            dummy_block,
```
Comment (gabe-l-hart) [inline tests/core/modules/test_module.py:200]:
```suggestion
            dummy_model,
```

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
Thanks for adding all of these! Let's change the word `block` to `model` everywhere since we've officially removed the notion of a `block` in `caikit`. Otherwise it looks good to go

**Review by gabe-l-hart — APPROVED** [review]
Looks great!

**Pull request merged**